Tools/environment_install/install-prereqs-windows.ps1

```diff
@@ -15,6 +15,7 @@ Start-Process -wait -FilePath $PSScriptRoot\setup-x86_64.exe -ArgumentList "--ro
 
 Write-Output "Downloading extra Python packages (4/5)"
 Start-Process -wait -FilePath "C:\cygwin64\bin\bash" -ArgumentList "--login -i -c 'pip2 install empy'"
+Start-Process -wait -FilePath "C:\cygwin64\bin\bash" -ArgumentList "--login -i -c 'ln -s /usr/bin/python2.7 /usr/bin/python'"
```

Comment:
need this to be "ln -sf" so if it exists we don't get an error

Reply:
we need to also check if /usr/bin/python exists ... when this is fixed in cygwin then we will be breaking with our install

Reply:
maybe "ln -s" is ok if it silently fails. Does it silently fail if the file exists?

Reply:
Hmm no, and I can't find a ln option that would fail silently.
But we could do ln -s  /usr/bin/python2.7 /usr/bin/python 2>/dev/null that will fail silently (although return code will still be non zero).

Reply:
Does cygwin provide Python3 by any chance?

Does *it* install a /usr/bin/python?

Reply:
Yes but not by default or  with the specific install we do.  

But if you had  cygwin prior and previously specifically asked for python 3.x then sure, but installed as /usr/bin/python3.x not /usr/bin/python.  So we should be ok here.

Reply:
That said maybe it did install python 3.x as /usr/bin/python in prior cygwin setups (as it used to for python 2.7).  But I don't know that testing for specific versions is worth it, or messing with waf. Users who've used cygwin and python 3 prior are probably rare and would know what to do.

Reply:
> nvm. ln -sf does work and fails silently.

Wait, what? "-f" should force-overwrite the target, so shouldn't fail, I'd have thought.

Reply:
You are right, my bad. I meant "is silent".  Or doesn't fail.  Or just gets it done no matter what, no fuss  :)

Reply:
We probably don't want to replace their Python3 link with a Python2 link, 'though.

Perhaps a one-liner isn't sufficient here.  Or maye just `2>/dev/null` and remove the `-f`?

Reply:
But we won't. Ill need to triple check, but my historical understanding is that cygwin installed as python only if python 2, not python 3, ever. So the only case python would be python3 is if user linked it or copied it themselves. The only thing that has changed is that neither are now installed as python.

In any case, it's easy to have more than a (bash) shell one liner, so we can be as sophisticated as we want here. Seems to me overkill but if you insist  :) ... check for python3? not sure how to do this atm but I doubt it be a big deal.

Reply:
> need this to be "ln -sf" so if it exists we don't get an error
> maybe "ln -s" is ok if it silently fails. Does it silently fail if the file exists?

@tridge Why do we care much about failing silently? That  corner case means the user either already has python 2,7 installed as python (ok) or python 3,x which user  manually linked prior. Might as well inform them.

Or perhaps something simple like this?
ln -s /usr/bin/python2.7 /usr/bin/python 2>/dev/null || echo "/usr/bin/python already exists: Please make sure it is python2.7 as Ardupilot builds with python 3.x will fail"